Source/SvgElement.cs

```diff
@@ -609,65 +619,81 @@ protected virtual void WriteAttributes(XmlTextWriter writer)
             var opacityValues = new Dictionary<string, float>();
 
-            foreach (var attr in _svgPropertyAttributes)
+            try
             {
-                if (attr.Property.Converter.CanConvertTo(typeof(string)))
-                {
-                    if (attr.Attribute.Name == "fill-opacity" || attr.Attribute.Name == "stroke-opacity")
-                    {
-                        opacityAttributes.Add(attr);
-                        continue;
-                    }
+                Inherited = false;
 
-                    if (!attr.Attribute.InAttributeDictionary || _attributes.ContainsKey(attr.Attribute.Name))
+                foreach (var attr in _svgPropertyAttributes)
+                {
+                    if (attr.Property.Converter.CanConvertTo(typeof(string)))
                     {
-                        var propertyValue = attr.Property.GetValue(this);
-
-                        var forceWrite = false;
-                        var writeStyle = attr.Attribute.Name == "fill" || attr.Attribute.Name == "stroke";
+                        if (attr.Attribute.Name == "fill-opacity" || attr.Attribute.Name == "stroke-opacity")
+                        {
+                            opacityAttributes.Add(attr);
+                            continue;
+                        }
 
-                        if (writeStyle && (Parent != null))
+                        if (!attr.Attribute.InAttributeDictionary || _attributes.ContainsKey(attr.Attribute.Name))
                         {
-                            if (propertyValue == SvgColourServer.NotSet) continue;
+                            var propertyValue = attr.Property.GetValue(this);
+
+                            var forceWrite = false;
+                            var writeStyle = attr.Attribute.Name == "fill" || attr.Attribute.Name == "stroke";
 
-                            object parentValue;
-                            if (TryResolveParentAttributeValue(attr.Attribute.Name, out parentValue))
+                            if (writeStyle && (Parent != null))
                             {
-                                if ((parentValue == propertyValue)
-                                    || ((parentValue != null) && parentValue.Equals(propertyValue)))
-                                    continue;
+                                if (propertyValue == SvgColourServer.NotSet) continue;
 
-                                forceWrite = true;
+                                object parentValue;
+                                if (TryResolveParentAttributeValue(attr.Attribute.Name, out parentValue))
+                                {
+                                    if ((parentValue == propertyValue)
+                                        || ((parentValue != null) && parentValue.Equals(propertyValue)))
+                                        continue;
+
+                                    forceWrite = true;
+                                }

```

Comment:
The compare tool really does a bad job here...

Reply:
I think so, too...